From: Gunnar Hjalmarsson <gunnarhj@ubuntu.com>
Subject: Use DejaVu as system font if language is Arabic

Several monospace fonts cause issues in VTE based terminals when
rendering Arabic scripts. DejaVu Sans Mono has proved to work best for
Arabic.

This patch sets DejaVu Sans Mono as the system font in case of an
Arabic or Persian locale if the dconf key "monospace-font-name" has
been assigned something else.

Bug-Ubuntu: https://launchpad.net/bugs/2002290
Bug-Debian: https://bugs.debian.org/1029707
Last-Update: 2023-02-15
Forwarded: not-needed

--- a/src/terminal-app.cc
+++ b/src/terminal-app.cc
@@ -1594,7 +1594,17 @@
 
   font = g_settings_get_string (app->desktop_interface_settings, MONOSPACE_FONT_KEY_NAME);
 
-  return pango_font_description_from_string (font);
+  auto desc = pango_font_description_from_string (font);
+
+  /* override system font for Arabic */
+  g_autofree gchar *lang = g_utf8_substring (g_get_language_names()[0], 0, 2);
+  if (g_str_equal (lang, "ar") || g_str_equal (lang, "fa")) {
+    auto family = pango_font_description_get_family (desc);
+    if (family && !g_str_has_prefix (family, "DejaVu Sans Mono"))
+      pango_font_description_set_family (desc, "DejaVu Sans Mono");
+  }
+
+  return desc;
 }
 
 #ifdef TERMINAL_SERVER
